src/agents/supervisor/agent.py

```diff
@@ -1,91 +1,298 @@
 import json
 from typing import Any, Dict, Literal, Protocol  # noqa UP
 
-from langchain_core.messages import AIMessage
+from langchain_core.exceptions import OutputParserException
+from langchain_core.messages import AIMessage, HumanMessage
 from langchain_core.output_parsers import PydanticOutputParser
 from langchain_core.prompts import ChatPromptTemplate, MessagesPlaceholder
 from langchain_core.pydantic_v1 import BaseModel, Field
 from langchain_core.runnables import RunnableSequence
+from langgraph.constants import END, START
+from langgraph.graph import StateGraph
+from langgraph.graph.graph import CompiledGraph
 
-from agents.common.constants import EXIT, FINALIZER
-from agents.common.state import AgentState
-from agents.common.utils import filter_messages
-from agents.supervisor.prompts import SUPERVISOR_ROLE_PROMPT, SUPERVISOR_TASK_PROMPT
+from agents.common.constants import (
+    FINALIZER,
+    MESSAGES,
+    NEXT,
+    PLANNER,
+)
+from agents.common.state import AgentState, Plan
+from agents.common.utils import create_node_output, filter_messages
+from agents.prompts import FINALIZER_PROMPT, PLANNER_PROMPT
+from agents.supervisor.prompts import ROUTER_ROLE_PROMPT, ROUTER_TASK_PROMPT
```

Comment:
Should we move the `FINALIZER_PROMPT`, `PLANNER_PROMPT` to `agents.supervisor.prompts`?

Reply:
yes!